Redesign model settings page with compact info and Advanced section

nickbaumann98 opened the pull request:
### Related Issue

<img width="464" height="528" alt="image" src="https://github.com/user-attachments/assets/93250aab-b261-4d34-b800-a8726660413f" />
<img width="452" height="655" alt="image" src="https://github.com/user-attachments/assets/f0edcdfb-57d7-4afe-a837-e1a9c6dc2294" />
<img width="467" height="701" alt="image" src="https://github.com/user-attachments/assets/0c8aba84-c939-4184-b548-f9c29906672a" />
<img width="469" height="857" alt="image" src="https://github.com/user-attachments/assets/740eb403-f339-46d7-8a29-bd0cc3dec4d3" />
<img width="326" height="74" alt="image" src="https://github.com/user-attachments/assets/fc876add-2fd4-40a5-9ac6-ce2e1506e4f8" />
<img width="303" height="87" alt="image" src="https://github.com/user-attachments/assets/60089f0f-490c-421c-b3c6-68a028a6a924" />
<img width="366" height="86" alt="image" src="https://github.com/user-attachments/assets/c6802fd4-db52-4ee8-9f49-5e1653e25720" />


**Issue:** N/A - UI/UX improvement for model settings page

### Description

This PR redesigns the model settings page to provide a cleaner, more compact interface:

**Changes:**
- **Compact pricing info**: Displays context window, input/output pricing in a single line (e.g., `Context: 200K | Input: $3/M | Output: $15/M`)
- **Capability badges with tooltips**: Visual badges for Images, Browser, and Caching support with hover tooltips explaining each capability
- **Collapsible Advanced section**: Groups cache pricing, tiered pricing, and provider routing under an expandable "Advanced" section
- **Removed redundancy**: Moved Provider Routing from standalone controls into Advanced section

All changes are backwards-compatible with existing provider components.

### Test Procedure

- Tested with Cline provider (OpenRouter models) to verify:
  - Compact info row displays correctly
  - Tooltips appear on hover for capability badges
  - Advanced section expands/collapses properly
  - Provider Routing dropdown functions correctly with dynamic description text
- Verified no breaking changes to other providers by checking that new props are optional
- Confirmed dropdown opens properly at bottom of Advanced section

### Type of Change

-   [x] 💅 Cosmetic Changes

### Pre-flight Checklist

-   [x] Changes are limited to a single feature, bugfix or chore
-   [ ] Tests are passing (`npm test`) and code is formatted and linted (`npm run format && npm run lint`)
-   [ ] I have created a changeset using `npm run changeset`
-   [x] I have reviewed contributor guidelines

### Screenshots

(Add before/after screenshots showing the compact layout and Advanced section)

### Additional Notes

This redesign makes the model settings interface more compact and scannable while preserving all functionality. The Advanced section keeps power-user features accessible without cluttering the primary interface.

<!-- ELLIPSIS_HIDDEN -->

----

> [!IMPORTANT]
> Redesigns model settings page for compact info and adds an Advanced section with provider routing, cache, and tiered pricing.
> 
>   - **UI Redesign**:
>     - `OpenRouterModelPicker.tsx`: Adds `showProviderRouting` prop to include provider routing in the Advanced section.
>     - `ModelInfoView.tsx`: Displays compact pricing info and capability badges with tooltips. Adds collapsible Advanced section for cache pricing, tiered pricing, and provider routing.
>     - `ClineProvider.tsx`: Uses `OpenRouterModelPicker` with `showProviderRouting` to include provider routing in the Advanced section.
>   - **Behavior**:
>     - Compact pricing info displayed in a single line.
>     - Capability badges for Images, Browser, and Caching with tooltips.
>     - Advanced section groups cache pricing, tiered pricing, and provider routing.
>   - **Testing**:
>     - Verified with Cline provider for correct display and functionality.
>     - Ensured no breaking changes to other providers by making new props optional.
> 
> <sup>This description was created by </sup>[<img alt="Ellipsis" src="https://img.shields.io/badge/Ellipsis-blue?color=175173">](https://www.ellipsis.dev?ref=cline%2Fcline&utm_source=github&utm_medium=referral)<sup> for 131104dae03e2d883548a9be7df30865daf8fb97. You can [customize](https://app.ellipsis.dev/cline/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>

<!-- ELLIPSIS_HIDDEN -->

arafatkatze commented on webview-ui/src/components/settings/providers/ClineProvider.tsx:
```diff
@@ -20,64 +16,24 @@ interface ClineProviderProps {
+			{/* View Billing & Usage Link */}
+			<div style={{ marginBottom: 12 }}>
+				<VSCodeLink href="https://openrouter.ai/activity" style={{ fontSize: "12px" }}>
+					View Billing & Usage
```
This block of billing and usage needs to be removed fully

arafatkatze commented on webview-ui/src/components/settings/providers/ClineProvider.tsx:
```diff
@@ -20,64 +16,24 @@ interface ClineProviderProps {
 			{showModelOptions && (
 				<>
-					{/* Provider Sorting Options */}
-					<VSCodeCheckbox
```
<img width="354" height="294" alt="Image" src="https://github.com/user-attachments/assets/6a3bb037-495d-4bf6-b130-5b7fce38c845" />

So the openrouter provider and cline provider should be consistent

nickbaumann98 commented:
Thanks for the review @arafatkatze. All feedback has been addressed:

> This block of billing and usage needs to be removed fully

Removed the duplicate "View Billing & Usage" link from ClineProvider - it's already in ClineAccountInfoCard.

> IF we remove provider sorting here then we should also get rid of it in Openrouter too for the sake of consistency
> So the openrouter provider and cline provider should be consistent

Unified provider routing across both providers - removed the inline checkbox+dropdown from OpenRouterProvider and both now use `showProviderRouting={true}` to render it in the Advanced section.

> Cache pricing precision

Fixed `formatCompactPrice` to show decimals when needed:
- `$3/M` for whole numbers
- `$3.75/M` for decimals

The only UX differences between Cline Provider and OpenRouter are now:
- Cline: Account Info Card with "Sign up with Cline"  
- OpenRouter: API key field and balance display


<img width="398" height="150" alt="image" src="https://github.com/user-attachments/assets/7a5438f5-dee2-4886-b6ab-2a5e6852e91a" />
<img width="393" height="173" alt="image" src="https://github.com/user-attachments/assets/ff00d2b5-d638-4c2a-a332-d64a6e6b22c2" />
<img width="416" height="780" alt="image" src="https://github.com/user-attachments/assets/4ff4fa3d-4a2f-4abd-b186-1a6822190fa4" />
